### PR TITLE
Dev/adding rds and jumpbox

### PR DESCRIPTION
**Pull request by ASVincent**
* probs need to add to the readme for the jumpbox and the rds instance.
* The underlying library used tags that started with a capital letter I think; might need to think about consistency with the rest of dea/ga.

### CODE REVIEW

File: ecs.tf [hunk 1]
```diff
@@ -27,27 +27,35 @@ module "ecs" {
   source = "modules/ecs"
 
   environment          = "${var.environment}"
-  cluster              = "${var.environment}"
-  cloudwatch_prefix    = "${var.environment}"          #See ecs_instances module when to set this and when not!
+  cluster              = "${var.vpc_name}"
```
Comment (tom-butler) [inline ecs.tf]: The standard we have used in other examples is stack_name, would that be a better fit here?

File: modules/ecs/main.tf [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
+  environment       = "${var.environment}"
+  enable_jumpbox    = "${var.enable_jumpbox}"
+  owner             = "${var.owner}"
+  key_name          = "${var.key_name}"
+  public_subnet_ids = "${module.network.public_subnet_ids}"
+}
+
 module "ecs_instances" {
-  source = "../ecs_instances"
+  source                  = "../ecs_instances"
```
Comment (tom-butler) [inline modules/ecs/main.tf]: Terraform has a built in function https://www.terraform.io/docs/commands/fmt.html  that will manage spacing for you. Most editors should have a terraform plugin that will call this for you on save. 

The terraform standard is to only align '=' when there is a grouping of at least 2 lines.

File: modules/jumpbox/jump-sg.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  tags {
+    Name        = "${var.cluster_name}-${var.environment}-jump_ssh"
+    Owner       = "${var.owner}"
+    Cluster     = "${var.cluster_name}"
+    Environment = "${var.environment}"
+    Created_by  = "terraform"
+  }
+}
+
+resource "aws_security_group" "ssh_from_jump_sg" {
```
Comment (tom-butler) [inline modules/jumpbox/jump-sg.tf]: This is a bit of legacy code that I'll be removing from the other examples soon, It's cleaner to just add the jumpbox rule (referencing the jumpbox security group) to the existing security group of a resource. This method requires a second security group for each resource, which can increase the chances of misconfiguring a security group.

File: modules/rds/rds-sg.tf [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#==============================================================
+
+# Security groups for the RDS.
+
+resource "aws_security_group" "rds" {
+  name        = "${var.cluster_name}_${var.environment}_ecs_rds_sg"
+  description = "allow traffic from the instance sg"
+  vpc_id      = "${var.vpc_id}"
+
+  ingress {
```
Comment (tom-butler) [inline modules/rds/rds-sg.tf:12]: Can we add a second ingress from the jumpbox?

File: modules/rds/variables.tf [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  description   = "Name of the first db"
+}
+
+variable "username" {
+  default       = "myuser"
+  description   = "User name"
+}
+
+variable "password" {
+  default       = "Password123"
```
Comment (tom-butler) [inline modules/rds/variables.tf]: Please remove this, we don't want users to accidentally set un secure passwords.

File: terraform.tfvars [hunk 6]
```diff
@@ -17,3 +19,9 @@ desired_capacity = 2
 instance_type = "t2.micro"
 
 ecs_aws_ami = "ami-c1a6bda2"
+
+owner = "alexander.vincent@ga.gov.au"
+
+ssh_ip_address = "192.104.44.129/32"
```
Comment (tom-butler) [inline terraform.tfvars]: Please remove this and inject it by setting the environment variable TF_VAR_ssh_ip_address.

File: ecs.tf [hunk 7]
```diff
@@ -27,27 +27,35 @@ module "ecs" {
...
+  cloudwatch_prefix    = "${var.vpc_name}-${var.environment}"          #See ecs_instances module when to set this and when not!
```
Comment (tom-butler) [inline ecs.tf]: Great Idea, that should make it much easier to debug.

File: terraform.tfvars [hunk 8]
```diff
@@ -17,3 +19,9 @@ desired_capacity = 2
 instance_type = "t2.micro"
 
 ecs_aws_ami = "ami-c1a6bda2"
+
+owner = "alexander.vincent@ga.gov.au"
```
Comment (tom-butler) [inline terraform.tfvars]: Good work adding the owner tag, we've historically run Janitor Monkey to clean up resources that don't have the Owner tag, as it makes it much easier to debug issues if you know who built the thing.

**Review by tom-butler — COMMENTED** [review]
Great work cleaning up the codebase.

**Comment by ASVincent** [comment]
I've implemented the feedback from the initial review.

File: modules/jumpbox/jump-sg.tf [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  name = "jump_ssh"
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["${var.ssh_ip_address}"]
+  }
+
+  ingress {
```
Comment (tom-butler) [inline modules/jumpbox/jump-sg.tf]: Why does the jumpbox need to ssh to itself?

Comment (ASVincent) [reply modules/jumpbox/jump-sg.tf]: I had a loop and I think I put that in because I misunderstood something; I've taken it out.

**Pull request merged**